common/environment/environment.cpp

```diff
@@ -1827,32 +1828,32 @@ CConstDropZoneServerInfoIterator::CConstDropZoneServerInfoIterator(const IConstD
 {
     Owned<IEnvironmentFactory> factory = getEnvironmentFactory();
     constEnv.setown((CLocalEnvironment *)factory->openEnvironment());
-    serverListIt.setown(dropZone->getPTree().getElements("ServerList"));
-    maxIndex = dropZone->getPTree().getCount("ServerList");
+    Owned<IPropertyTree> pSrc = &dropZone->getPTree();
+    serverListIt.setown(pSrc->getElements("ServerList"));
+    maxIndex = pSrc->getCount("ServerList");
 }
 
 bool CConstDropZoneServerInfoIterator::first()
 {
     bool hasFirst = serverListIt->first();
     if (hasFirst)
-        curr = new CConstDropZoneServerInfo(constEnv, &serverListIt->query());
+        curr.setown(new CConstDropZoneServerInfo(constEnv, &serverListIt->query()));
```

Comment:
Added the clear() into the else branch but if there is no first then 'curr' can have any value which should clear?